### PR TITLE
PR for #4761: Added missing peerDependencies

### PR DESCRIPTION
**Pull request by nate-summercook**
Added missing peerDependencies in order to not break projects using yarn pnp, issue #4761.

### CODE REVIEW

File: packages/x-date-pickers/package.json [hunk 1]
```diff
@@ -55,6 +55,9 @@
     "rifm": "^0.12.1"
   },
   "peerDependencies": {
+    "@babel/runtime": "^7.16.3",
```
Comment (flaviendelangle) [inline packages/x-date-pickers/package.json:58]: On the core, `@babel/runtime` is a dep, not a peer dep

I wonder why we never needed it on the data grid packages...
@mnajdova do you know what it is being used for on the core ?

**Review by flaviendelangle — COMMENTED** [review]
Thanks for your contribution !

File: packages/x-date-pickers/package.json (hunk 1 above)
Comment (nate-summercook) [reply packages/x-date-pickers/package.json:58]: Would work for me too to put `@babel/runtime` into dependencies. But it needs to be defined somewhere.

Comment (nate-summercook) [reply packages/x-date-pickers/package.json]: Looks like most libraries would put that into the dependencies section. Went ahead and changed it in the PR. Also I made the emotion imports optional, as mui can also be used with styled-components.

**Comment by nate-summercook** [comment]
Maybe give your TimePicker and DateTimePickers (or actually all date related components) a fix time in your tests, so they don't fail randomly ;)

**Comment by flaviendelangle** [comment]
> Maybe give your TimePicker and DateTimePickers (or actually all date related components) a fix time in your tests, so they don't fail randomly ;)

We did, but it broke clearly
I must spend some time investigating...

**Review by flaviendelangle — APPROVED** [review]
We will need to do the same for emotion on the datagrid packages
I would like to know why we need this babel plugin on the pickers and not on the data grid

But for the pickers, the modification make sense.
I will also do the same on `@mui/x-date-pickers-pro` in a follow up PR :+1:

**Comment by flaviendelangle** [comment]
> have you tried my reproduction approach (with yarn 3 & pnp) with the datagrid?

No, I will do it before trying to apply a fix :+1: 
I am just surprised nobody raised it before. But maybe yarn 3 & pnp is not used a lot yet.

File: packages/x-date-pickers/package.json [hunk 2]
```diff
@@ -55,6 +56,8 @@
     "rifm": "^0.12.1"
   },
   "peerDependencies": {
+    "@emotion/react": "^11.9.0",
+    "@emotion/styled": "^11.8.1",
```
Comment (oliviertassinari) [inline packages/x-date-pickers/package.json:60]: Outside of the scope that I think should have been as low as possible, I would personally assume that this change is going in the wrong direction: #32668.

**Pull request merged**